cmd/kube-apiserver/app/options/validation.go

```diff
@@ -79,6 +80,21 @@ func (options *ServerRunOptions) Validate() []error {
 	if options.MasterCount <= 0 {
 		errors = append(errors, fmt.Errorf("--apiserver-count should be a positive number, but value '%d' provided", options.MasterCount))
 	}
+
+	// TODO: remove when we stop supporting the legacy group version.
+	if options.APIEnablement != nil {
+		for key, value := range options.APIEnablement.RuntimeConfig {
+			if strings.HasPrefix(key, "v1") || strings.HasPrefix(key, "api/v1") {
+				delete(options.APIEnablement.RuntimeConfig, key)
```

Comment:
validation is not the right place for this. Wondering where we can place this.

Reply:
Actually, I find it is a better place than anywhere. Maybe I missed something, I am not sure.

Reply:
https://github.com/kubernetes/kubernetes/pull/57999 ;-)

Reply:
For this special case we could move it to https://github.com/sttts/kubernetes/blob/d030c2d8e51a21585d6879b0b4a0beeaca5181ca/cmd/kube-apiserver/app/server.go#L119, with a good comment why. I am not a fan of mutation in a validate func.

Reply:
SGTM

Reply:
Compare https://github.com/kubernetes/kubernetes/pull/57999#issuecomment-356239143